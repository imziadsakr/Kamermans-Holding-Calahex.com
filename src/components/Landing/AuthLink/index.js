import React from 'react' ;

import { Link } from 'react-router-dom';

import {
    Box ,
    Button 
} from '@mui/material' ;

import { makeStyles } from '@mui/styles';

const useStyles =  makeStyles((theme) => ({
    root : {
        textAlign : "center",
        "& .MuiButtonBase-root" : {
            boxShadow: '0px 3px 1px -2px rgb(0 0 0 / 20%), 0px 2px 2px 0px rgb(0 0 0 / 14%), 0px 1px 5px 0px rgb(0 0 0 / 12%)' ,
            borderRadius : "20px" ,
            marginLeft : "10px" ,
            marginRight : "10px" ,
        } ,
        "& .MuiButton-contained" : {
            textTransform : "captialize !important"
        },
        "& a" :{
            textDecoration : "none" ,
        }
    },
    signup : {
        color : "white"
    },
    login : {
        color : theme.palette.primary.main
    }
}));

const AuthLink = () => {
    
    const classes = useStyles() ;

    return (
        <Box className={classes.root}>
            <Button variant='contained' size='large' >
                <Link to="/signup" >
                    <Box className={classes.signup}>
                        Sign Up
                    </Box>
                </Link>
            </Button>
            <Button variant='outlined' size='large'>
                <Link to={'/login'}>
                    <Box className={classes.login}>
                        Log In
                    </Box>
                </Link>
            </Button>
        </Box>
    )
}

export default AuthLink ;