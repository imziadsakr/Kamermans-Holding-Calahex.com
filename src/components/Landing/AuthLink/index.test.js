import React from 'react';
import renderer from 'react-test-renderer';

import theme from '../../../utils/theme';
import { ThemeProvider } from '@mui/material';

import { BrowserRouter } from 'react-router-dom';

import AuthLink from './index' ;

describe("AuthLink", () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <ThemeProvider theme={theme}>
                                <AuthLink />
                    </ThemeProvider>
                </BrowserRouter>    
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})