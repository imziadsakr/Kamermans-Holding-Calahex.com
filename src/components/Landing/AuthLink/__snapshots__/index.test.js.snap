// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLink renders correctly 1`] = `
<div
  className="makeStyles-root-1 MuiBox-root css-0"
>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-ksofmu-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <a
      href="/signup"
      onClick={[Function]}
    >
      <div
        className="makeStyles-signup-2 MuiBox-root css-0"
      >
        Sign Up
      </div>
    </a>
  </button>
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root  css-r6008z-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      <div
        className="makeStyles-login-3 MuiBox-root css-0"
      >
        Log In
      </div>
    </a>
  </button>
</div>
`;
